Update deprecated API uses

roystgnr opened the pull request:
Refs #13921

## Reason

C++ `public:` access to the `libMesh::MeshBase::boundary_info` member was deprecated in https://github.com/libMesh/libmesh/pull/2763 last year, and the old `libMesh::UniquePtr` shim class was deprecated in https://github.com/libMesh/libmesh/pull/2910 last month, but we still had a couple files using these old APIs in the MOOSE modules.

## Design

Use the `get_boundary_info()` accessor method in the first case, and good old `std::unique_ptr` in the second.

## Impact

This lets MOOSE compile against a `--disable-deprecated` libMesh build again.

roystgnr committed:
Use get_boundary_info() accessor, not raw member

The `public:` access to that member is deprecated.

Refs #13921

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/745933/) on fbccadd wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/17914/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/17914/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format c5a8ae7c6b6ffecff6abecb83c6093943e378b0e`

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit 27141d18f3137f77e33cdb3d565fd38ebfbfc46f
+Subproject commit c02af017673655633edfc4ebf8454f46bd94f059
```
**Caution!** This contains a submodule update

roystgnr replied:
Wow, that whole "no unresolved conversations" thing has already paid off.  I did not actually intend to get a submodule change caught up in this PR.

loganharbour commented:
@roystgnr feel like making precheck happy?

roystgnr committed:
clang-format update

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/750604/) on bfff176 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/17914/site)

This comment will be updated on new commits.

roystgnr committed:
libMesh::UniquePtr -> std::unique_ptr

We offered this shim for years thanks to some lousy compilers but it's
deprecated now.

Refs #13921
